source/agora/consensus/validation/Block.d

```diff
@@ -112,6 +118,268 @@ public string isInvalidReason (const ref Block block, in ulong prev_height,
     return null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a genesis block
+
+    Follow the Block rules except for the following
+        - Block height is 0
+        - The previous block hash of the block must be empty
+        - The number of transactions in the block is above zero and less than
+          Block.TxsInBlock
+        - All the transactions pass validation, which implies:
+            - Transaction type in block should only be Payment or Freeze
+            - There is empty a input for the transaction
+            - There is one or more outputs of the transaction
+        - All the enrollments pass validation, which implies:
+            - The enrollments refer to freeze tx's in this block
+
+    Params:
+        block = The genesis block to check
+
+    Returns:
+        `null` if the genesis block is valid, a string explaining the reason it
+        is invalid otherwise.
```

Comment:
I know you probably didn't write this, there is another one of these somewhere else, but it's not proper English. This is the right way:

```suggestion
        `null` if the genesis block is valid, otherwise a string explaining the reason it
        is invalid.
```